internal/pkg/search/search.go

```diff
@@ -56,6 +56,17 @@ func ConfigsForTemplateInstance(all []*model.ConfigWithRows, instanceID string)
 	return matches
 }
 
+// ConfigsByTemplateInstance group configurations by the instance ID.
+func ConfigsByTemplateInstance(all []*model.ConfigWithRows) map[string][]*model.ConfigWithRows {
+	out := make(map[string][]*model.ConfigWithRows)
+	for _, object := range all {
+		if instanceID := object.Metadata.InstanceID(); instanceID != "" {
+			out[instanceID] = append(out[instanceID], object)
+		}
+	}
+	return out
+}
+
```

Comment:
There is a slice with all configurations in the project, and we need to group them by the `instanceId`.